client/gutenberg/editor/calypsoify-iframe.tsx

```diff
@@ -202,17 +238,13 @@ class CalypsoifyIframe extends Component {
 			// sending the new MessageChannel from the server
 			this.iframePort.postMessage( {
 				action: 'loadRevision',
-				payload: {
-					title: revision.post_title,
-					excerpt: revision.post_excerpt,
-					content: revision.post_content,
-				},
+				payload: { title, excerpt, content },
 			} );
 		}
 	};
 
-	closeMediaModal = media => {
-		if ( ! this.state.classicBlockEditorId && media ) {
+	closeMediaModal = ( media: { items: Array< Parameters< typeof mediaCalypsoToGutenberg > > } ) => {
+		if ( ! this.state.classicBlockEditorId && media && this.mediaSelectPort ) {
```

Comment:
The `this.mediaSelectPort` check is interesting. Did the typechecker force you to add it? Because it shouldn't be needed, for the following reasons:
- `closeMediaModal` can be called only when the Media Modal is rendered, i.e., `this.state.isMediaModalVisible` is true
- `this.state.isMediaModalVisible` and `this.mediaSelectPort` are set and unset in sync, when opening and closing the modal.

The fact that one of the properties is set on instance and the other on state, and that the state update can be scheduled for later rather than done immediately, looks like a weak point of the program.

Also, the `isMediaModalVisible` property is redundant, as it can be derived from `this.mediaSelectPort`.

Is it possible to reorg the program in such a way that the typechecker "knows" the above facts? That would be better use of the type system than just adding checks that are not really needed. It's similar to patching ESLint a11y warnings in such a way that the linter shuts up, but the actual a11y remains bad.

Reply:
> Did the typechecker force you to add it?

yes it did. you make a valid claim about it and it's a case where I like to argue with the type checker. to its credit though, we occasionally get code refactors that change an assumption that currently holds. in these cases it provides a guard against unintended side-effects of other changes.

> Is it possible to reorg the program in such a way that the typechecker "knows" the above facts?

This is great! It's an advanced type of type-driven design that seeks to remove as much information as is possible; in this case it removes the opportunity for `this.mediaSelectPort` and `isMediaModalVisible` to get out of sync.

I may try and get this change in outside of this PR as it seems like a good one. I'll double-check though to be sure.

Reply:
good idea @jsnajdr but I don't think we can simplify yet. the classic block opens the media modal but doesn't pass a `MessagePort` so in the case that we open the media model from the classic block `this.mediaSelectPort` will actually be `null`